app/src/behaviors/behavior_backlight.c

```diff
@@ -20,21 +20,59 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 
 static int behavior_backlight_init(const struct device *dev) { return 0; }
 
+static int
+on_keymap_binding_convert_central_state_dependent_params(struct zmk_behavior_binding *binding,
+                                                         struct zmk_behavior_binding_event event) {
+    switch (binding->param1) {
+    case BL_TOG_CMD: {
+        bool state;
+        int err = zmk_backlight_get_state(&state);
+        if (err) {
+            LOG_ERR("Failed to get backlight state (err %d)", err);
+            return err;
+        }
+
+        binding->param1 = state ? BL_OFF_CMD : BL_ON_CMD;
+        break;
+    }
+    case BL_INC_CMD: {
+        uint8_t brightness = zmk_backlight_calc_brt(1);
```

Comment:
This logic doesn't work, check this comment: https://github.com/zmkfirmware/zmk/pull/904#discussion_r689291345

Reply:
It worked on my board when I was testing it

Reply:
It works most of the time but there are some edge cases, there are many examples in the original PR comments. I'll try to add some tests to detect those corner cases.

Reply:
Looking through that comment I cannot see what edge cases mentioned won't get caught within my Inc and Dec functions. It's very close to the currently implemented RGB underglow behaviours API

Reply:
I'm sorry, I misunderstood your code. Your code works fine but you have a different interpretation than mine.
I assume that pressing `BL_INC` when `state.on = false`, turns on the backlight, check my code: [https://github.com/bortoz/zmk/blob/backlight/app/src/backlight.c#L154](https://github.com/bortoz/zmk/blob/backlight/app/src/backlight.c#L154). There was also some thought about that in the discussion I have liked.
I honestly don't remember why I made that choice, however I think this implementation is fine too.

Reply:
yes pressing BL_INC when it's off turns on the backlight, the `zmk_backlight_calc_brt` function limits the backlight to within the range of the max value, and `state.on` is calculated automatically